contracts/DissentVoting.sol

```diff
@@ -0,0 +1,388 @@
+        }
+
+        uint256 totalVotes = vote_.yea.add(vote_.nay);
+
+        // Vote ended?
+        if (_isVoteOpen(vote_)) {
+            return false;
+        }
+        // Has enough support?
+        if (!_isValuePct(vote_.yea, totalVotes, vote_.supportRequiredPct)) {
+            return false;
+        }
+        // Has min quorum?
+        if (!_isValuePct(vote_.yea, vote_.votingPower, vote_.minAcceptQuorumPct)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    function getVote(uint256 _voteId)
+        public
+        view
+        voteExists(_voteId)
+        returns (
+            bool open,
+            bool executed,
+            uint64 startDate,
+            uint64 snapshotBlock,
+            uint64 supportRequired,
+            uint64 minAcceptQuorum,
+            uint256 yea,
+            uint256 nay,
+            uint256 votingPower,
+            bytes script
+        )
+    {
+        Vote storage vote_ = votes[_voteId];
+
+        open = _isVoteOpen(vote_);
+        executed = vote_.executed;
+        startDate = vote_.startDate;
+        snapshotBlock = vote_.snapshotBlock;
+        supportRequired = vote_.supportRequiredPct;
+        minAcceptQuorum = vote_.minAcceptQuorumPct;
+        yea = vote_.yea;
+        nay = vote_.nay;
+        votingPower = vote_.votingPower;
+        script = vote_.executionScript;
+    }
+
+    function getVoterState(uint256 _voteId, address _voter) public view voteExists(_voteId) returns (VoterState) {
+        return votes[_voteId].voters[_voter];
+    }
+
+    function _newVote(bytes _executionScript, string _metadata, bool _castVote, bool _executesIfDecided)
+        internal
+        returns (uint256 voteId)
+    {
+        uint256 votingPower = token.totalSupplyAt(vote_.snapshotBlock);
+        require(votingPower > 0, ERROR_NO_VOTING_POWER);
+
+        voteId = votesLength++;
+        Vote storage vote_ = votes[voteId];
+        vote_.startDate = getTimestamp64();
+        vote_.snapshotBlock = getBlockNumber64() - 1; // avoid double voting in this very block
+        vote_.supportRequiredPct = supportRequiredPct;
+        vote_.minAcceptQuorumPct = minAcceptQuorumPct;
+        vote_.votingPower = votingPower;
+        vote_.executionScript = _executionScript;
+
+        emit StartVote(voteId, msg.sender, _metadata);
+
+        if (_castVote && canVote(voteId, msg.sender)) {
+            _vote(voteId, true, msg.sender, _executesIfDecided);
+        }
+    }
+
+    function _vote(
+        uint256 _voteId,
+        bool _supports,
+        address _voter,
+        bool _executesIfDecided
+    ) internal
+    {
+        Vote storage vote_ = votes[_voteId];
+
+        // This could re-enter, though we can assume the governance token is not malicious
+        uint256 voterStake = token.balanceOfAt(_voter, vote_.snapshotBlock);
+
+        //don't have to decrease voterStake as the original voting-app as canExecute() checks if _voter has already voted
```

Comment:
Maybe:

"Removed resetting of vote stake, that was in the original voting-app, as voters can only vote once."